pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,141 @@
+            $`"name:${.table_name}` ++
+            $`","unique:${cond {(.unique = 'UNIQUE'):'true', _:'false'}}` ++
+            $`","null_filtered:${cond {(.nullfiltered = 'NULL_FILTERED'):'true', _:'false'}}` ++
+            $`","key_parts:${//seq.join(', ', .key_part orderby .)}` ++
+            cond {.storing_col: $`","storing:' ${//seq.join(', ',.storing_col)}`}  ++
+            cond {.interleaved_table: $`","interleave_in:' ${.interleaved_table}`})}"]`,
+    };
+
+# allEntityPatterns generates the pattern string for an entity
+let allEntityPatterns = \entity \model
+    $`${entityPatternDelimiter(entity, model, '[')}${cond {
+        hasEntityPattern(entity, model):
+        //seq.join(', ',[entityPattern(entity, model, 0), entityPattern(entity, model, 1),
+        entityPattern(entity, model, 2), entityPattern(entity, model, 3)] where .@item != '')
+    }}${entityPatternDelimiter(entity, model, ']')}`;
+
+# allAttributePatterns generates the pattern string for an entity attribute
+let allAttributePatterns = \entity \attr
+    $`${attributePatternDelimiter(entity,attr,'[')}${cond {
+        hasAttributePattern(entity,attr):
+        //seq.join(', ',[attributePattern(entity,attr,2), attributePattern(entity,attr,0), attributePattern(entity,attr,1),
+        attributePattern(entity,attr,3), attributePattern(entity,attr,4)] where .@item != '')
+    }}${attributePatternDelimiter(entity,attr,']')}`;
+
+# typeInfo generates the type info for an attribute
+let typeInfo = \entity \attr \type \isArray
+    let fkAttr = entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_attribute orderby .) where .;
+    let fkTable = entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_table orderby .) where .;
+    cond {
+        fkAttr | fkTable:  //seq.join('', (fkTable => //seq.join('', .) orderby .)) ++ '.'
+            ++ //seq.join('', (fkAttr => //seq.join('', .) orderby .)),
+        isArray: $`sequence of ${type}`,
+        _: type,
+    };
+
+# transformModel translates the empty model into sysl file
+let transformModel = \model \package
+    # sysl specification
+    # https://github.com/anz-bank/sysl/blob/master/pkg/sysl/sysl.proto
+    $`
+        ########   THIS IS AUTOGENERATED BY sysl   ########
+        ${model.schema => $`
+            ${.name}${cond {package: $`[package="${package}"]`} }:
+            ${//seq.sub('#>>>\n', '')($`
+            ${(model.entities => \entity $`
+            #>>>
+                !table ${entity.name} ${allEntityPatterns(entity, model)}:
+                    ${entity.attributes >>
+                        $`
+                            ${.name} <: ${typeInfo(entity, ., .type ++ size(.length), .array)}${cond {.nullable:'?'}} ${allAttributePatterns(entity, .)}` ::\i:
+                        }`
+                    )
+                    orderby .::\i:
+            }
+            `)}
+        ` orderby .:::}
```

Comment:
Nit: you only need `::` unless you're planning to append something to the end.

```suggestion
        ` orderby .::}
```

Reply:
fine